Tests/MongoSwiftSyncTests/MongoCollectionTests.swift

```diff
@@ -256,6 +256,31 @@ final class MongoCollectionTests: MongoSwiftTestCase {
         expect(try self.coll.deleteOne(["cat": "cat"])?.deletedCount).to(equal(1))
     }
 
+    func testDeleteOneWithHint() throws {
+        try withTestNamespace { client, _, collection in
+            let requirements = TestRequirement(
+                minServerVersion: ServerVersion(major: 4, minor: 4)
+            )
+            let unmetRequirement = try client.getUnmetRequirement(requirements)
+            guard unmetRequirement == nil else {
+                printSkipMessage(testName: self.name, unmetRequirement: unmetRequirement!)
+                return
+            }
+
+            let doc1: BSONDocument = ["_id": 1, "cat": "meow", "dog": "bark"]
+            let doc2: BSONDocument = ["_id": 2, "cat": "nya", "dog": "woof"]
+            try collection.insertMany([doc1, doc2])
+
+            let model = IndexModel(keys: ["cat": 1, "dog": 1])
+            expect(try collection.createIndex(model)).to(equal("cat_1_dog_1"))
+
+            let opts1 = DeleteOptions(hint: ["cat": 1, "dog": 1])
+            let opts2 = DeleteOptions(hint: "cat_1_dog_1")
+            expect(try collection.deleteOne(["cat": "meow"], options: opts1)?.deletedCount).to(equal(1))
+            expect(try collection.deleteOne(["cat": "nya"], options: opts2)?.deletedCount).to(equal(1))
```

Comment:
can we add an assertion that a bad hint fails for each of these?

Reply:
added for `deleteOne` and `deleteMany`